Add constants for string literals in UI test classes

anusreelakshmi934 opened the pull request:
Fixes #1153

vaisakhkannan commented on src/main/java/io/openliberty/tools/intellij/util/Constants.java:
```diff
@@ -94,4 +94,21 @@ public final class Constants {
+    public static final String SERVICE_LIVE_HEALTH_CHECK_JAVA = "ServiceLiveHealthCheck.java";
+    public static final String MPG_PROPERTIES = "microprofile-config.properties";
+
+    public static final String CLOSE_ALL_TABS = "Close All Tabs";
```
@anusreelakshmi934 , based on Trevor suggestion, since constants are only used for the tests, then they should go in a file under the test directory.

anusreelakshmi934 replied:
I have modified it. Please re review.

vaisakhkannan commented on src/test/java/io/openliberty/tools/intellij/it/TestUtils.java:
```diff
@@ -541,7 +542,7 @@ public static boolean isServerStopNeeded(String wlpInstallPath) {
     public static void checkDebugPort(String absoluteWLPPath, int debugPort) throws IOException {
         // Retrieve the WLP server.env file path
-        Path serverEnvPath = Paths.get(absoluteWLPPath, "wlp", "usr", "servers", "defaultServer", "server.env");
+        Path serverEnvPath = Paths.get(absoluteWLPPath, "wlp", "usr", "servers", "defaultServer", ItConstants.SERVER_ENV);
```
@TrevCraw , Do we need to consider adding all those strings, such as 'wlp', 'usr', 'servers', etc., to the `ITConstants class`? if not then we can keep "server.env" here instead of `ItConstants.SERVER_ENV`

TrevCraw replied:
I don't think it would be beneficial to create a constant for each individual string. However, there is a lot of repetition when defining the various `Path` variables. It would be beneficial to create constants that contain a section of a path, for instance `wlp/usr/servers/defaultServer`, which could then be re-used elsewhere.

dessina-devasia reviewed (commented):
Change Copyright year in all files

anusreelakshmi934 commented on src/test/java/io/openliberty/tools/intellij/it/Utils/ItConstants.java:
```diff
@@ -0,0 +1,22 @@
+package io.openliberty.tools.intellij.it.Utils;
```
Added the Copyrights header in ItConstants.java file

anusreelakshmi934 commented:
> Change Copyright year in all files

Changed the Copyright year

mrglavas reviewed (changes requested):
@anusreelakshmi934 I see several new failures that are occurring on the Linux and Windows builds, for example:

```
GradleSingleModJakartaLSTest > initializationError FAILED
    java.lang.RuntimeException: Unable to open file SystemResource
        at io.openliberty.tools.intellij.it.UIBotTestUtils.openFile(UIBotTestUtils.java:700)
        at io.openliberty.tools.intellij.it.SingleModJakartaLSTestCommon.prepareEnv(SingleModJakartaLSTestCommon.java:204)
        at io.openliberty.tools.intellij.it.GradleSingleModJakartaLSTest.setup(GradleSingleModJakartaLSTest.java:40)

        Caused by:
        java.util.NoSuchElementException: List is empty.
            at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:221)
            at com.intellij.remoterobot.fixtures.Fixture.findText(Fixture.kt:46)
            at io.openliberty.tools.intellij.it.UIBotTestUtils.openFile(UIBotTestUtils.java:687)
            ... 2 more
```
I suspect these might be related to recent changes in computing paths in this PR. Please verify.

dessina-devasia reviewed (approved):
Looks good to me. Thanks!

anusreelakshmi934 commented:
> @anusreelakshmi934 I see several new failures that are occurring on the Linux and Windows builds, for example:
> 
> ```
> GradleSingleModJakartaLSTest > initializationError FAILED
>     java.lang.RuntimeException: Unable to open file SystemResource
>         at io.openliberty.tools.intellij.it.UIBotTestUtils.openFile(UIBotTestUtils.java:700)
>         at io.openliberty.tools.intellij.it.SingleModJakartaLSTestCommon.prepareEnv(SingleModJakartaLSTestCommon.java:204)
>         at io.openliberty.tools.intellij.it.GradleSingleModJakartaLSTest.setup(GradleSingleModJakartaLSTest.java:40)
> 
>         Caused by:
>         java.util.NoSuchElementException: List is empty.
>             at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:221)
>             at com.intellij.remoterobot.fixtures.Fixture.findText(Fixture.kt:46)
>             at io.openliberty.tools.intellij.it.UIBotTestUtils.openFile(UIBotTestUtils.java:687)
>             ... 2 more
> ```
> 
> I suspect these might be related to recent changes in computing paths in this PR. Please verify.

Hi @mrglavas . I have made the changes and now the test succeeds.

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModJakartaLSTestCommon.java:
```diff
@@ -198,10 +199,10 @@ public static void prepareEnv(String projectPath, String projectName) {
-        UIBotTestUtils.openFile(remoteRobot, projectName, "SystemResource", projectName, "src", "main", "java", "io.openliberty.mp.sample", "system");
-        UIBotTestUtils.openFile(remoteRobot, projectName, "SystemResource2", projectName, "src", "main", "java", "io.openliberty.mp.sample", "system");
+        UIBotTestUtils.openFile(remoteRobot, projectName, ItConstants.SYSTEM_RESOURCE, projectName, "src", "main", "java", "io.openliberty.mp.sample", "system");
+        UIBotTestUtils.openFile(remoteRobot, projectName, ItConstants.SYSTEM_RESOURCE_2, projectName, "src", "main", "java", "io.openliberty.mp.sample", "system");
```
Why not use `SYSTEM_DIR_PATH` for these two lines?

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java:
```diff
@@ -591,16 +592,16 @@ public static void prepareEnv(String projectPath, String projectName) {
 
         // open server.xml file
-        UIBotTestUtils.openFile(remoteRobot, projectName, "server.xml", projectName, "src", "main", "liberty", "config");
+        UIBotTestUtils.openFile(remoteRobot, projectName, ItConstants.SERVER_XML, projectName, "src", "main", "liberty", "config");
```
Why not use the `CONFIG_DIR_PATH` constant here (and below)?

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModMPLSTestCommon.java:
```diff
@@ -344,7 +345,7 @@ public static void prepareEnv(String projectPath, String projectName) {
 
         UIBotTestUtils.openFile(remoteRobot, projectName, "ServiceLiveHealthCheck", projectName, "src", "main", "java", "io.openliberty.mp.sample", "health");
-        UIBotTestUtils.openFile(remoteRobot, projectName, "microprofile-config.properties", projectName, "src", "main", "resources", "META-INF");
+        UIBotTestUtils.openFile(remoteRobot, projectName, ItConstants.MPG_PROPERTIES, projectName, "src", "main", "resources", "META-INF");
```
Why not use `META_INF_DIR_PATH` here?

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/TestUtils.java:
```diff
@@ -27,7 +28,7 @@ public class TestUtils {
      * WLP messages.log path.
      */
-    public static final Path WLP_MSGLOG_PATH = Paths.get("wlp", "usr", "servers", "defaultServer", "logs", "messages.log");
+    public static final Path WLP_MSGLOG_PATH = Paths.get(ItConstants.DEFAULT_SERVER_PATH, "logs", "messages.log");
```
It would be good to create a constant for the `messages.log` path.

anusreelakshmi934 replied:
modified the code.

TrevCraw reviewed (changes requested):
It looks like a few spots to use the new constants may have been missed. I only marked the spots visible in the GitHub diffs. Please make sure to search through the test code to see if there are any other spots that were missed where the new constants can be used or where new constants should be created.

anusreelakshmi934 commented:
> It looks like a few spots to use the new constants may have been missed. I only marked the spots visible in the GitHub diffs. Please make sure to search through the test code to see if there are any other spots that were missed where the new constants can be used or where new constants should be created.

Hi @TrevCraw . Initially, I replaced all occurrences where the new constants could be used. However, this caused a GitHub Actions (GHA) failure where the file could not be opened. Upon investigation, I discovered that the `openFile` method expects the `filePath` parameter in the format `String... filePath`. Replacing it with the constant resulted in an error because the input was not provided in the required `String...` format.

I also tried passing the value as a `String` array, but this led to a compilation issue. The problem arises because the method expects multiple strings as individual arguments and does not accept an array in place of these arguments.

Expected Output is  - 
![image](https://github.com/user-attachments/assets/572e0e84-3830-4593-a0b7-f5c05b83845c)

But if we pass the constant we will get - 
![image](https://github.com/user-attachments/assets/754f2b2e-0bf7-4db7-a5b9-b9c726c10717)

If we pass the value as String array we will get - 
<img width="773" alt="image" src="https://github.com/user-attachments/assets/68abf25f-60f5-4dab-b14b-1dc8e1522176" />

So the file will not be opened. Hence I kept the strings in openFile as it was.

TrevCraw commented:
> If we pass the value as String array we will get - <img alt="image" width="773" src="https://private-user-images.githubusercontent.com/141303561/403283307-68abf25f-60f5-4dab-b14b-1dc8e1522176.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5ODI2MDYsIm5iZiI6MTczNjk4MjMwNiwicGF0aCI6Ii8xNDEzMDM1NjEvNDAzMjgzMzA3LTY4YWJmMjVmLTYwZjUtNGRhYi1iMTRiLTFkYzhlMTUyMjE3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNVQyMzA1MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMGU1ZTM4YzgyODdkODIwMDM4NTEyZmY5ZjM1NjdlYTM2MTMyN2U1NjU5ZDhkZTJjZjEyM2QwYTQxMDEzMjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rvkRnRzt5LgjTuI6HGdB6ne_r74QeXsFoSOY6KpJf6A">
> 
> So the file will not be opened. Hence I kept the strings in openFile as it was.

Hi @anusreelakshmi934 , looking at the image above, it looks like you used `Arrays.toString()` for your `String` array parameter. That will not work, as `Arrays.toString()` returns a `String`, not an array of `Strings`. You should be able to pass an array of `Strings` to the `String...` varargs parameter.

If you take a step back and look at the original code, you will noticed the pattern for both `getPath()` and `openFile()` was always set up to work with `String...`. We should leave that format alone and adjust the newly created constants to match the existing code patterns. It would probably be best to modify your `String` constants that are paths with slashes to be `String` arrays instead.

anusreelakshmi934 commented:
> > If we pass the value as String array we will get - <img alt="image" width="773" src="https://private-user-images.githubusercontent.com/141303561/403283307-68abf25f-60f5-4dab-b14b-1dc8e1522176.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5ODI2MDYsIm5iZiI6MTczNjk4MjMwNiwicGF0aCI6Ii8xNDEzMDM1NjEvNDAzMjgzMzA3LTY4YWJmMjVmLTYwZjUtNGRhYi1iMTRiLTFkYzhlMTUyMjE3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNVQyMzA1MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMGU1ZTM4YzgyODdkODIwMDM4NTEyZmY5ZjM1NjdlYTM2MTMyN2U1NjU5ZDhkZTJjZjEyM2QwYTQxMDEzMjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rvkRnRzt5LgjTuI6HGdB6ne_r74QeXsFoSOY6KpJf6A">
> > So the file will not be opened. Hence I kept the strings in openFile as it was.
> 
> Hi @anusreelakshmi934 , looking at the image above, it looks like you used `Arrays.toString()` for your `String` array parameter. That will not work, as `Arrays.toString()` returns a `String`, not an array of `Strings`. You should be able to pass an array of `Strings` to the `String...` varargs parameter.
> 
> If you take a step back and look at the original code, you will noticed the pattern for both `getPath()` and `openFile()` was always set up to work with `String...`. We should leave that format alone and adjust the newly created constants to match the existing code patterns. It would probably be best to modify your `String` constants that are paths with slashes to be `String` arrays instead.

Thanks, @TrevCraw, for the clarification. Initially, when I passed the string array to the `openFile` method, it was being interpreted as the fifth parameter, with the project name as the fourth. This caused issues with passing the string array. To resolve this, I modified the approach by combining the `projectName` and the path. With this change, two of the classes are now passing, but others are still failing despite using the same approach. I am currently investigating the issue and will update the PR accordingly.

mrglavas commented:
> Thanks, @TrevCraw, for the clarification. Initially, when I passed the string array to the `openFile` method, it was being interpreted as the fifth parameter, with the project name as the fourth. This caused issues with passing the string array. To resolve this, I modified the approach by combining the `projectName` and the path. With this change, two of the classes are now passing, but others are still failing despite using the same approach. I am currently investigating the issue and will update the PR accordingly.

@anusreelakshmi934 I know you're currently working on updates to this PR so will wait for that before reviewing again. For the current state of the PR, I had similar feedback to what @TrevCraw already expressed. It should be possible to construct arrays from a combination of constants and variable values that are equivalent to the vararg syntax (for `String...`) that was in the original code.

mrglavas reviewed (changes requested):
See my recent comments. Awaiting changes before reviewing again.

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/Utils/ItConstants.java:
```diff
@@ -0,0 +1,40 @@
+
+    public static final String[] CONFIG_DIR_PATH = { "src", "main", "liberty", "config" };
+    public static final String[] SYSTEM_DIR_PATH = { "src", "main", "java", "io.openliberty.mp.sample", "system"};
+    public static final String[] HEALTH_DIR_PATH = { "src", "main", "java", "io.openliberty.mp.sample", "health" };
```
Why use dots as part of the String values in the String arrays `SYSTEM_DIR_PATH` and `HEALTH_DIR_PATH`? The original hard coded values do not use any dots in most cases. Now you have to convert all the dots to slashes. Why not just remove the dots from the constants so you do not have to do that conversion each time?

Note, there are some spots - like https://github.com/OpenLiberty/liberty-tools-intellij/blob/main/src/test/java/io/openliberty/tools/intellij/it/SingleModJakartaLSTestCommon.java#L201 - where dots have to be used because of the way the UI robot expects the values. For these cases, we can use a different constant.

TrevCraw reviewed (changes requested):
There are lots of places where `String.join()` is used. Would it be possible to pass String arrays as was done before? You can build the String arrays from your existing constants dynamically if needed. Or, in most cases, you can create new constants that combine your existing constants.

For example, you have:
```
CONFIG_DIR_PATH = { "src", "main", "liberty", "config" };
SERVER_XML = "server.xml";
```

You can create a new constant that combines the two called `SERVER_XML_PATH` and then use that to pass as a parameter to the necessary methods.

anusreelakshmi934 commented:
Hi @TrevCraw, 
I’ve addressed your comments and made the necessary changes. Please review them when you get a chance and let me know if this aligns with your expectations or if further modifications are needed. Thanks! 

Additionally, I’ve removed the [WLP_MSGLOG_PATH](https://github.com/OpenLiberty/liberty-tools-intellij/blob/034fa100e27273a30a07772d2aa550c01032b3e0/src/test/java/io/openliberty/tools/intellij/it/TestUtils.java#L30) from the TestUtils file and moved it to the ItConstants file as a slash-separated path. I felt that, instead of calling `Paths.get()`, it might be more efficient to directly specify the path there. Could you confirm if this approach is appropriate?

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModJakartaLSTestCommon.java:
```diff
@@ -90,8 +91,8 @@ public void testInsertJakartaCodeSnippetIntoJavaPart() {
-            UIBotTestUtils.insertCodeSnippetIntoSourceFile(remoteRobot, "SystemResource.java", snippetStr, snippetChooser);
-            Path pathToSrc = Paths.get(projectsPath, projectName, "src", "main", "java", "io", "openliberty", "mp", "sample", "system", "SystemResource.java");
+            UIBotTestUtils.insertCodeSnippetIntoSourceFile(remoteRobot, ItConstants.SYSTEM_RESOURCE_JAVA, snippetStr, snippetChooser);
+            Path pathToSrc = Paths.get(projectsPath, projectName, ItConstants.SYSTEM_DIR_PATH, ItConstants.SYSTEM_RESOURCE_JAVA);
```
Why is SYSTEM_DIR_PATH a String and not a String[]? It would be better to keep the format the same as the original code. You can always build a new String[] constant from String[] SYSTEM_DIR_PATH and String SYSTEM_RESOURCE_JAVA.

anusreelakshmi934 replied:
I updated the `SYSTEM_DIR_PATH` to a `String[]`. Additionally, I introduced new variables: `SERVER_XML_PATH`, `SERVER_ENV_PATH`, and `BOOTSTRAP_PROPERTIES_PATH`.

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModLibertyLSTestCommon.java:
```diff
@@ -122,15 +123,15 @@ public void testInsertFeatureIntoServerXML() {
         try {
             UIBotTestUtils.insertStanzaInAppServerXML(remoteRobot, stanzaSnippet, 18, 40, UIBotTestUtils.InsertionType.FEATURE, true);
-            Path pathToServerXML = Paths.get(projectsPath, projectName, "src", "main", "liberty", "config", "server.xml");
+            Path pathToServerXML = Paths.get(projectsPath, projectName, ItConstants.CONFIG_DIR_PATH, ItConstants.SERVER_XML);
```
Same comment here as for SYSTEM_DIR_PATH above. Would be better to keep as a String[].

anusreelakshmi934 replied:
I have updated the SYSTEM_DIR_PATH to a String[]

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModMPLSTestCommon.java:
```diff
@@ -95,8 +97,8 @@ public void testInsertCodeSnippetIntoJavaPart() {
-            UIBotTestUtils.insertCodeSnippetIntoSourceFile(remoteRobot, "ServiceLiveHealthCheck.java", snippetStr, snippetChooser);
-            Path pathToSrc = Paths.get(projectsPath, projectName, "src", "main", "java", "io", "openliberty", "mp", "sample", "health","ServiceLiveHealthCheck.java");
+            UIBotTestUtils.insertCodeSnippetIntoSourceFile(remoteRobot, ItConstants.SERVICE_LIVE_HEALTH_CHECK_JAVA, snippetStr, snippetChooser);
+            Path pathToSrc = Paths.get(projectsPath, projectName, ItConstants.HEALTH_DIR_PATH, ItConstants.SERVICE_LIVE_HEALTH_CHECK_JAVA);
```
Same comment here as for SYSTEM_DIR_PATH and CONFIG_DIR_PATH above. Would be better to keep as a String[].

anusreelakshmi934 replied:
I have updated the SYSTEM_DIR_PATH to a String[]

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModMPLSTestCommon.java:
```diff
@@ -201,14 +203,14 @@ public void testInsertMicroProfileProperty() {
-            UIBotTestUtils.insertConfigIntoMPConfigPropertiesFile(remoteRobot, "microprofile-config.properties", cfgSnippet, cfgNameChooserSnippet, cfgValueSnippet, true);
-            Path pathToMpCfgProperties = Paths.get(projectsPath, projectName, "src", "main", "resources", "META-INF", "microprofile-config.properties");
+            UIBotTestUtils.insertConfigIntoMPConfigPropertiesFile(remoteRobot, ItConstants.MPG_PROPERTIES, cfgSnippet, cfgNameChooserSnippet, cfgValueSnippet, true);
+            Path pathToMpCfgProperties = Paths.get(projectsPath, projectName, String.join(File.separator, ItConstants.META_INF_DIR_PATH), ItConstants.MPG_PROPERTIES);
```
Why do a String.join() here? The original code is passing a String[].

anusreelakshmi934 replied:
I have removed the String.join() and combined the projectName and MPCFG path using combinePath method.

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/SingleModMPLSTestCommon.java:
```diff
@@ -294,19 +296,19 @@ public void testQuickFixInMicroProfileConfigProperties() {
         String expectedHoverData = "Type mismatch: boolean expected. By default, this value will be interpreted as 'false'";
 
-        Path pathToMpCfgProperties = Paths.get(projectsPath, projectName,"src", "main", "resources", "META-INF", "microprofile-config.properties");
+        Path pathToMpCfgProperties = Paths.get(projectsPath, projectName, String.join(File.separator, ItConstants.META_INF_DIR_PATH), ItConstants.MPG_PROPERTIES);
```
Same as above comment on String.join().

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/TestUtils.java:
```diff
@@ -541,7 +540,7 @@ public static boolean isServerStopNeeded(String wlpInstallPath) {
     public static void checkDebugPort(String absoluteWLPPath, int debugPort) throws IOException {
         // Retrieve the WLP server.env file path
-        Path serverEnvPath = Paths.get(absoluteWLPPath, "wlp", "usr", "servers", "defaultServer", "server.env");
+        Path serverEnvPath = Paths.get(absoluteWLPPath, ItConstants.DEFAULT_SERVER_PATH, ItConstants.SERVER_ENV);
```
Same comment here as for SYSTEM_DIR_PATH, CONFIG_DIR_PATH and HEALTH_DIR_PATH above. Would be better to keep as a String[].

anusreelakshmi934 replied:
I have updated it to String[].

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/Utils/ItConstants.java:
```diff
@@ -0,0 +1,43 @@
+    public static final String COMPACT_MODE = "Compact Mode";
+
+    public static final String SERVICE_LIVE_HEALTH_CHECK_JAVA = "ServiceLiveHealthCheck.java";
+    public static final String MPG_PROPERTIES = "microprofile-config.properties";
```
```suggestion
    public static final String MPCFG_PROPERTIES = "microprofile-config.properties";
```

TrevCraw commented on src/test/java/io/openliberty/tools/intellij/it/TestUtils.java:
```diff
@@ -60,7 +59,7 @@ public static void validateLibertyServerStopped(String testName, String wlpInsta
         printTrace(TraceSevLevel.INFO, testName + ":validateLibertyServerStopped: Entry.");
 
-        String wlpMsgLogPath = Paths.get(wlpInstallPath, WLP_MSGLOG_PATH.toString()).toString();
+        String wlpMsgLogPath = Paths.get(wlpInstallPath, ItConstants.MESSAGES_LOG_PATH).toString();
```
Since `wlpInstallPath` is a String and you have now changed `MESSAGES_LOG_PATH` to a String, there is no reason to use `Paths.get()` to convert the values to a Path and then convert back to a String with `toString()`. May as well just concatenate the Strings together to build the full path as a String.

anusreelakshmi934 replied:
I have modified the code accordingly

TrevCraw replied:
I do not see these changes in the latest diff

anusreelakshmi934 replied:
Hi @TrevCraw . Sorry for the confusion. I have made the MESSAGES_LOG_PATH a String[] similar to all the other constants. Thus I have kept it as Paths.get().

TrevCraw reviewed (changes requested):
Hi @anusreelakshmi934 , please address the review comments below. And be sure to check the build results after making your changes.

It probably was not necessary to move `MESSAGES_LOG_PATH` to the `ItConstants` file since it is only used in one file. However, the change should be OK. Please take a look at one of the comments related to this change below.

turkeylurkey commented on src/test/java/io/openliberty/tools/intellij/it/SingleModJakartaLSTestCommon.java:
```diff
@@ -12,6 +12,7 @@
 import com.automation.remarks.junit5.Video;
 import com.intellij.remoterobot.RemoteRobot;
 import com.intellij.remoterobot.fixtures.JTreeFixture;
+import io.openliberty.tools.intellij.it.Utils.ItConstants;
```
I'd like to suggest using `import static io.openliberty.tools.intellij.it.Utils.ItConstants.*` so that you do not have to specify the class name every time you use a constant in all these test files. It gets repetitive.

vaisakhkannan reviewed (approved):
Looks good

vaisakhkannan reviewed (changes requested):
@anusreelakshmi934 ,I think it's better to define constants for string literals present in the child test class as well.

anusreelakshmi934 commented on src/test/java/io/openliberty/tools/intellij/it/GradleSingleModJakartaLSTest.java:
updated the header

vaisakhkannan commented on src/test/java/io/openliberty/tools/intellij/it/GradleSingleModMPProjectTest.java:
```diff
@@ -43,8 +45,8 @@ public static void setup() {
         setSmMpProjResURI("api/resource");
         setSmMPProjOutput("Hello! Welcome to Open Liberty");
```
Okay, since we agreed to address this in new follow-up issue, we can proceed with merging this PR.

vaisakhkannan reviewed (dismissed):
looks good

vaisakhkannan reviewed (dismissed):
approving again
